### PR TITLE
Add lozalization to more-info-vacuum

### PR DESCRIPTION
**Pull request by edenhaus**
## Proposed change

- Converted more-info-vacuum to Lit
- Introduced VacuumCommand, so we can use a stream to create the command icons
- Add localization to more-info-vacuum

### Not translated
I have not translated `stateObj.attributes.status`, because this attribute can be any value.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
vacuum:
  - platform: demo
```

## Additional information

I'm not sure if introducing the new class VacuumCommand was correct, but in my opinion it makes the code more clear and understandable.
I have separated each proposed into a separated commit, so it should be easier to see the different changes.
As this is my first pullrequest in this project, feel free to changes or to give me feedback :)

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @edenhaus, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant-polymer#4793).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/dialogs/more-info/controls/more-info-vacuum.ts [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+  VACUUM_SUPPORT_LOCATE,
+  VACUUM_SUPPORT_PAUSE,
+  VACUUM_SUPPORT_RETURN_HOME,
+  VACUUM_SUPPORT_START,
+  VACUUM_SUPPORT_STATUS,
+  VACUUM_SUPPORT_STOP,
+  VacuumEntity,
+} from "../../../data/vacuum";
+
+class VacuumCommand {
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-vacuum.ts]: Seems unnecessary to create a class for this, can we just make it an typed object?

File: src/dialogs/more-info/controls/more-info-vacuum.ts [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+    const supportsReturnHome = supportsFeature(
+      stateObj,
+      VACUUM_SUPPORT_RETURN_HOME
+    );
+    const supportsFanSpeed = supportsFeature(
+      stateObj,
+      VACUUM_SUPPORT_FAN_SPEED
+    );
+    const supportsBattery = supportsFeature(stateObj, VACUUM_SUPPORT_BATTERY);
+    const supportsStatus = supportsFeature(stateObj, VACUUM_SUPPORT_STATUS);
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-vacuum.ts]: Since we use these only once, it is nice to inline the check instead of creating a const.

**Review by bramkragten — COMMENTED** [review]
Looks great! Thanks. Just a few small remarks.

File: src/dialogs/more-info/controls/more-info-vacuum.ts (hunk 2 above)
Comment (edenhaus) [reply src/dialogs/more-info/controls/more-info-vacuum.ts]: Changed it to inline, except for `supportsCommandBar` as in my opinion it is clearer if it is a variable and not checked inline.

File: src/dialogs/more-info/controls/more-info-vacuum.ts (hunk 1 above)
Comment (edenhaus) [reply src/dialogs/more-info/controls/more-info-vacuum.ts]: I changed the class to an interface. I was not sure if this is your intention, but I'm new in typescript and this I found on google for typed object. If not, please give me an example

**Comment by edenhaus** [comment]
The button `pause` and `play_pause` had the same tooltip "Pause" (before this PR). I changed the tooltip of `play_pause` to "Start/Pause"

File: src/dialogs/more-info/controls/more-info-vacuum.ts (hunk 2 above)
Comment (bramkragten) [reply src/dialogs/more-info/controls/more-info-vacuum.ts]: Actually, I only meant the ones that were used once, the ones that are used twice (in `supportsCommandBar`) should be a `const`.

Comment (edenhaus) [reply src/dialogs/more-info/controls/more-info-vacuum.ts]: Today I had another idea, I can use `VACUUM_COMMANDS.some(...)` so we don't have any duplicate code and the maintenance of the code is easier.

File: src/dialogs/more-info/controls/more-info-vacuum.ts [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+                </span>
+                <span><strong>${stateObj.attributes.status}</strong></span>
+              </div>
+            `
+          : ""}
+        ${supportsFeature(stateObj, VACUUM_SUPPORT_BATTERY)
+          ? html`
+              <div">
+                <span>
+                  <iron-icon icon=${stateObj.attributes.battery_icon}></iron-icon>
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-vacuum.ts]:
```suggestion
                  <iron-icon .icon=${stateObj.attributes.battery_icon}></iron-icon>
```

File: src/dialogs/more-info/controls/more-info-vacuum.ts [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+                )}
+              </div>
+              <div class="flex-horizontal">
+                ${VACUUM_COMMANDS.filter((item) =>
+                  item.isVisible(stateObj)
+                ).map(
+                  (item) => html`
+                    <div>
+                      <paper-icon-button
+                        icon=${item.icon}
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-vacuum.ts]:
```suggestion
                        .icon=${item.icon}
```

File: src/dialogs/more-info/controls/more-info-vacuum.ts [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+                        .entry=${item}
+                        @click=${this.callService}
+                        title=${this.hass!.localize(
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-vacuum.ts]:
```suggestion
                        .title=${this.hass!.localize(
```

File: src/dialogs/more-info/controls/more-info-vacuum.ts [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+              </div>
+            </div>
+          `
+        : ""}
+      ${supportsFeature(stateObj, VACUUM_SUPPORT_FAN_SPEED)
+        ? html`
+            <div>
+              <div class="flex-horizontal">
+                <ha-paper-dropdown-menu
+                  label=${this.hass!.localize(
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-vacuum.ts]:
```suggestion
                  .label=${this.hass!.localize(
```

File: src/dialogs/more-info/controls/more-info-vacuum.ts [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+                >
+                  <paper-listbox
+                    slot="dropdown-content"
+                    .selected=${stateObj.attributes.fan_speed}
+                    @iron-select=${this.handleFanSpeedChanged}
+                    attr-for-selected="item-name"
+                  >
+                    ${stateObj.attributes.fan_speed_list!.map(
+                      (mode) => html`
+                        <paper-item item-name=${mode}>
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-vacuum.ts]:
```suggestion
                        <paper-item .itemName=${mode}>
```

File: src/dialogs/more-info/controls/more-info-vacuum.ts [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+  private callService(ev: CustomEvent) {
+    const entry = (ev.target! as any).entry as VacuumCommand;
+    this.hass.callService("vacuum", entry.serviceName, {
+      entity_id: this.stateObj!.entity_id,
+    });
+  }
+
+  private handleFanSpeedChanged(ev: CustomEvent) {
+    const oldVal = this.stateObj!.attributes.fan_speed;
+    const newVal = ev.detail.item.getAttribute("item-name");
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-vacuum.ts]:
```suggestion
    const newVal = ev.detail.item.itemName);
```

**Comment by edenhaus** [comment]
Thanks for your suggestions :)

**Review by bramkragten — APPROVED** [review]

**Pull request merged**